File: Configuration/src/main/java/com/medecineWebApp/Configuration/models/Services.java
package com.medecineWebApp.Configuration.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.io.Serializable;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "services")
@EntityListeners({AuditingEntityListener.class,AuditLogListener.class})
public class Services extends Auditable implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
   // @JsonProperty("nom")
    private String name;
    private String description;
    private Long medicalRecordId;

    @ManyToOne
    @JoinColumn(name = "departement_id", nullable = false)
    @JsonBackReference
    private Departement departement;


}

File: Client/frontend/src/app/models/services.ts
import { IDepartement } from "./departments";

export interface IServices{
    id?:number;
    name?:string;
    description?:string;
    medicalRecordId?:number;
    departement?:IDepartement;

}

export class Services implements IServices{
    constructor(
    public id?:number,
    public name?:string,
    public description?:string,
    public medicalRecordId?:number,
    public departement?:IDepartement
    ){}

}